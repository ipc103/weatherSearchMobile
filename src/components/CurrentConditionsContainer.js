import React, { Component } from 'react'
import { View } from 'react-native'

import { fetchCurrentConditions } from '../api/weatherUnderground'
import CurrentConditionsDetail from './CurrentConditionsDetail'
import SearchForm from './SearchForm'

class CurrentConditionsContainer extends Component {
  constructor(){
    super()
    this.state = {
      city: '',
      region: '',
      conditions: {
        message: 'Enter a city and state abbreviation'
      }
    }
  }

  handleQueryChange = (query) => {
    this.setState(query)
  }

  handleSearch = (event) => {
    event.preventDefault()
    fetchCurrentConditions(this.state.city, this.state.region)
      .then(this.parseWeather)
  }

  parseWeather = (data) => {
    if (data.current_observation) {
      this.setState({
        city: '',
        region: '',
        conditions: {
          temp: data.current_observation.temperature_string,
          weather: data.current_observation.weather.toLowerCase(),
          city: `${data.location.city}, ${data.location.state}`
        }
      })
    } else {
      this.setState({
        conditions: {
          message: 'No results found'
        }
      })
    }
  }

  render(){
    return(
      <View>
        <SearchForm handleQueryChange={this.handleQueryChange} city={this.state.city} region={this.state.region} handleSearch={this.handleSearch}/>
        <CurrentConditionsDetail conditions={this.state.conditions} />
      </View>
    )
  }
}

export default CurrentConditionsContainer
