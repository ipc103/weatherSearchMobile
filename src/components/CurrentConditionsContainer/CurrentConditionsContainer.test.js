import React from 'react'
import renderer from 'react-test-renderer';

import CurrentConditionsContainer from './'

describe('CurrentConditionsContainer', () => {
  const wrapper = renderer.create(<CurrentConditionsContainer />)

  it('renders without crashing', () => {
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  describe("handling search query change", () => {

    it('updates the city', () => {
      const instance = wrapper.root.instance
      instance.handleQueryChange({city: 'Ann Arbor'})
      expect(instance.state.city).toEqual('Ann Arbor')
    })
  })
})
