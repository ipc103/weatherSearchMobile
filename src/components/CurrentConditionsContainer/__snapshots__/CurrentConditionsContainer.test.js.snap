// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentConditionsContainer renders without crashing 1`] = `
<View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Enter City
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      style={
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 40,
        }
      }
      value=""
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Enter State
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      style={
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 40,
        }
      }
      value=""
    />
    <View
      accessibilityComponentType="button"
      accessibilityLabel="Learn more about this purple button"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#841584",
              },
            ]
          }
        >
          Learn More
        </Text>
      </View>
    </View>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Enter a city and state abbreviation
    </Text>
  </View>
</View>
`;
